app/src/main/java/com/medmapper/v33001/MainActivity.kt

```diff
@@ -20,17 +20,19 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : ComponentActivity() {
 
-    private var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
     private val viewModel: MainViewModel by viewModel<MainViewModel>()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        val firebaseUser = FirebaseAuth.getInstance().currentUser
+        firebaseUser?.let {
+            val user = User(it.uid, "")
+            viewModel.user = user
+            viewModel.listenToMedicine()
+        }
+
         setContent {
-            firebaseUser?.let {
-                val user = User(it.uid, "")
-                viewModel.user = user
-                viewModel.listenToMedicine()
-            }
```

Comment:
I'd leave the firebaseUser declared at class scope (as an attribute), instead of the onCreate() method scope.  Keeping it as an attribute is useful because:

- It persists login information between sessions.  In other words, the user can close and open the app, and does not need to log in again.
- As an attribute, the user variable is available to all functions in the class.  You generally do want to keep variables as narrowly scoped as possible, but this is one case where you likely want to use the user attribute across multiple functions.

I don't see the if test that you mention is removed with the null safe check (?.)